pandas-stubs/io/json/_json.pyi

```diff
@@ -11,194 +9,89 @@ from pandas.core.series import Series
 
 from pandas._typing import (
     CompressionOptions,
-    FilePathOrBuffer,
-    JSONSerializable,
+    DtypeArg,
+    FilePath,
+    HashableT,
+    ReadBuffer,
+    StorageOptions,
 )
 
-loads = ...
-dumps = ...
-
-def to_json(
-    path_or_buf,
-    obj,
-    orient: str | None = ...,
-    date_format: str = ...,
-    double_precision: int = ...,
-    force_ascii: bool = ...,
-    date_unit: str = ...,
-    default_handler: Callable[[Any], JSONSerializable] | None = ...,
-    lines: bool = ...,
-    compression: str | None = ...,
-    index: bool = ...,
-    indent: int = ...,
-): ...
-
-class Writer:
-    obj = ...
-    orient = ...
-    date_format = ...
-    double_precision = ...
-    ensure_ascii = ...
-    date_unit = ...
-    default_handler = ...
-    index = ...
-    indent = ...
-    is_copy = ...
-    def __init__(
-        self,
-        obj,
-        orient: str | None,
-        date_format: str,
-        double_precision: int,
-        ensure_ascii: bool,
-        date_unit: str,
-        index: bool,
-        default_handler: Callable[[Any], JSONSerializable] | None = ...,
-        indent: int = ...,
-    ) -> None: ...
-    def write(self): ...
-
-class SeriesWriter(Writer): ...
-class FrameWriter(Writer): ...
-
-class JSONTableWriter(FrameWriter):
-    schema = ...
-    obj = ...
-    date_format = ...
-    orient = ...
-    index = ...
-    def __init__(
-        self,
-        obj,
-        orient: str | None,
-        date_format: str,
-        double_precision: int,
-        ensure_ascii: bool,
-        date_unit: str,
-        index: bool,
-        default_handler: Callable[[Any], JSONSerializable] | None = ...,
-        indent: int = ...,
-    ): ...
-
 @overload
 def read_json(
-    path: FilePathOrBuffer,
-    orient: str | None = ...,
-    dtype=...,
-    convert_axes=...,
-    convert_dates: bool = ...,
+    path_or_buf: FilePath | ReadBuffer[str] | ReadBuffer[bytes],
+    *,
+    orient: Literal["split", "records", "index", "columns", "values", "table"]
+    | None = ...,
+    typ: Literal["series", "frame"] = ...,
+    dtype: bool | dict[HashableT, DtypeArg] | None = ...,
+    convert_axes: bool | None = ...,
+    convert_dates: bool | list[str] = ...,
     keep_default_dates: bool = ...,
-    numpy: bool = ...,
     precise_float: bool = ...,
-    date_unit: str | None = ...,
+    date_unit: Literal["s", "ms", "us", "ns"] | None = ...,
     encoding: str | None = ...,
+    encoding_errors: Literal[
+        "strict", "ignore", "replace", "backslashreplace", "surrogateescape"
+    ]
+    | None = ...,
     lines: bool = ...,
-    chunksize: int | None = ...,
+    chunksize: int,
```

Comment:
according to docs, `chunksize` can only be passed if `lines==True`, so we might as well force the type of `lines` to be `True` in this overload

Reply:
Fixed.